### PR TITLE
Use v1.0.0 release of trusted services 

### PR DESCRIPTION
**Pull request by gowthamsk-arm**
The PR includes the following changes:

1. Updates to docker files to use v1.0.0 of trusted services
2. Configure the trusted service provider to use new RPC APIs
3. Disable the key mappings test for the TS provider

Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>

### CODE REVIEW

**Comment by gowthamsk-arm** [comment]
Manual CI run for testing new docker images:
https://github.com/parallaxsecond/parsec/actions/runs/9029129762

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -180,18 +180,21 @@ jobs:
     needs: [build-and-export-cross-compile-docker]
     steps:
       - uses: actions/checkout@v3
-      - name: Load Docker
-        uses: ./.github/actions/load_docker
-        if: ${{ env.TEST_CROSS_DOCKER_IMAGE == 'parsec-service-test-cross-compile' }}
-        with:
-          image-name: "${{ env.TEST_CROSS_DOCKER_IMAGE }}"
-          image-path: "/tmp"
+      # - name: Load Docker
```
Comment (tgonzalezorlandoarm) [inline .github/workflows/ci.yml:183]: Why is this commented out? Should it be removed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -180,18 +180,21 @@ jobs:
...
       - name: Run the cross compiler tests using pre-built docker image
-        if: ${{ env.TEST_CROSS_DOCKER_IMAGE != 'parsec-service-test-cross-compile' }}
-        run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec ghcr.io/parallaxsecond/parsec-service-test-cross-compile /tmp/parsec/test/cross-compile.sh
-      - name: Run the cross compiler tests using image built on the CI
         if: ${{ env.TEST_CROSS_DOCKER_IMAGE == 'parsec-service-test-cross-compile' }}
-        run: docker run -v $(pwd):/tmp/parsec -w "${{ env.TEST_CROSS_DOCKER_IMAGE }}" /tmp/parsec/test/cross-compile.sh
+        run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec -t parsec-service-test-cross-compile /tmp/parsec/test/cross-compile.sh
+      - name: Run the cross compiler tests using image built on the CI
+        if: ${{ env.TEST_CROSS_DOCKER_IMAGE != 'parsec-service-test-cross-compile' }}
+        run: docker run -v $(pwd):/tmp/parsec -w /tmp/parsec -t parsec-service-test-cross-compile /tmp/parsec/test/cross-compile.sh
```
Comment (tgonzalezorlandoarm) [inline .github/workflows/ci.yml:197]: This is the same as line 194?

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile [hunk 3]
```diff
@@ -161,3 +161,18 @@ ENV SPIFFE_ENDPOINT_SOCKET="unix:///tmp/agent.sock"
 
 # Add safe.directory configuration to access repos freely
 RUN git config --global --add safe.directory '*'
+
+# Install latest Trusted Services libraries. The previously installed
+# libraries are old and necessary for ./generate-keys.sh which uses 
+# Parsec 1.0.0 version that is incompatible with newer libts APIs. 
+RUN rm /usr/local/lib/libts.so* /usr/local/lib/libprotobuf-nanopb.a /usr/local/lib/libmbedcrypto.a
```
Comment (tgonzalezorlandoarm) [inline e2e_tests/docker_image/parsec-service-test-all.Dockerfile:168]: Why are these removed?

File: build.rs [hunk 4]
```diff
@@ -19,6 +19,7 @@ fn generate_ts_bindings(ts_include_dir: String) -> Result<()> {
     println!("cargo:rerun-if-changed={}", header);
 
     let bindings = bindgen::Builder::default()
+        .clang_arg(format!("-I{}", ts_include_dir))
```
Comment (tgonzalezorlandoarm) [inline build.rs:22]: What is this for?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tgonzalezorlandoarm) [reply .github/workflows/ci.yml:197]: (Why?)

**Review by tgonzalezorlandoarm — CHANGES_REQUESTED** [review]

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile [hunk 5]
```diff
@@ -161,3 +161,18 @@ ENV SPIFFE_ENDPOINT_SOCKET="unix:///tmp/agent.sock"
 
 # Add safe.directory configuration to access repos freely
 RUN git config --global --add safe.directory '*'
+
+# Install latest Trusted Services libraries. The previously installed
+# libraries are old and necessary for ./generate-keys.sh which uses 
+# Parsec 1.0.0 version that is incompatible with newer libts APIs. 
+RUN rm /usr/local/lib/libts.so* /usr/local/lib/libprotobuf-nanopb.a /usr/local/lib/libmbedcrypto.a 
+RUN git clone https://git.trustedfirmware.org/TS/trusted-services.git --branch integration \
```
Comment (anta5010) [inline e2e_tests/docker_image/parsec-service-test-all.Dockerfile:169]: In the latest Yocto TS 1.0 updates the branch was changed from integration to main:
https://git.yoctoproject.org/meta-arm/tree/meta-arm/recipes-security/trusted-services/trusted-services-src.inc

Comment (anta5010) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:169]: You might also need to update [trusted-services-vendor] to the main branch

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gowthamsk-arm) [reply .github/workflows/ci.yml:183]: The last patch is only for testing purpose to build and test cross compiler docker image on the CI. It will be dropped after the CI is green.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (gowthamsk-arm) [reply .github/workflows/ci.yml:197]: This patch will be dropped. This is only for testing purpose.

File: build.rs (hunk 4 above)
Comment (gowthamsk-arm) [reply build.rs:22]: To access rpc_caller_session.h header file.

Comment (gowthamsk-arm) [reply build.rs:22]: This file provides the new RPC calls.

Comment (gowthamsk-arm) [reply build.rs:22]: This file provides the new RPC calls.

Comment (gowthamsk-arm) [reply build.rs:22]: This file provides the new RPC calls.

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile (hunk 3 above)
Comment (gowthamsk-arm) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:168]: These are old libraries that are used for Parsec 1.0.0.

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile (hunk 5 above)
Comment (gowthamsk-arm) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:169]: Thanks for the comment Anton. Will update this.

**Comment by gowthamsk-arm** [comment]
New CI run: https://github.com/parallaxsecond/parsec/actions/runs/9031610348

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile (hunk 3 above)
Comment (tgonzalezorlandoarm) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:168]: Why are they there in the first place?

Comment (anta5010) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:168]: Please have a look at the docker file from the beginning. Initially we build an old version of Parsec with an old TS libraries. After generate-keys.sh script is run we build a new version of Parsec with a new version of TS libraries.

**Review by ionut-arm — APPROVED** [review]
Looks good, I only have one minor comment

File: ci.sh [hunk 6]
```diff
@@ -120,7 +120,7 @@ run_old_e2e_tests() {
 run_key_mappings_tests() {
     # There is no keys generated for CryptoAuthLib yet.
```
[the remark this answers is not in the document]
Comment (gowthamsk-arm) [reply ci.sh:121]: Yes will add a comment for it.

**Review by ionut-arm — APPROVED** [review]
🚀

**Review by tgonzalezorlandoarm — APPROVED** [review]
